lang/src/accounts/option.rs

```diff
@@ -0,0 +1,77 @@
+//! Option<T> type for optional accounts.
+//!
+//! # Example
+//! ```ignore
+//! #[derive(Accounts)]
+//! pub struct Example {
+//!     pub my_acc: Option<Account<'info, MyData>>
+//! }
+//! ```
+
+use std::collections::{BTreeMap, BTreeSet};
+
+use solana_program::account_info::AccountInfo;
+use solana_program::instruction::AccountMeta;
+use solana_program::pubkey::Pubkey;
+
+use crate::{Accounts, AccountsClose, AccountsExit, Result, ToAccountInfos, ToAccountMetas};
+
+impl<'info, T: Accounts<'info>> Accounts<'info> for Option<T> {
```

Comment:
This optionality functionality will break though right if someone tries to `option<>` wrap a nested `derive(Accounts)` struct which contains more than a single account on it?

Reply:
Ok, looking at this with rust analyzer, the mutable reference is also passed onto the sub `Account<DataAccount>` which isn't what i expected but good that it works.